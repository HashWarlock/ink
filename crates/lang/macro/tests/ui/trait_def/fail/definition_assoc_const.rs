use ink_lang as ink;

#[ink::trait_definition]
pub trait WithAssocType {
    const INVALID_CONST: bool;

    #[ink(message)]
    fn message(&self);
}

fn main() {}
