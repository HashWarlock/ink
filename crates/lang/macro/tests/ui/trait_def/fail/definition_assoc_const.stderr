error: associated constants in ink! trait definitions are not supported, yet
 --> $DIR/definition_assoc_const.rs:5:5
  |
5 |     const INVALID_CONST: bool;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
